app/main.py

```diff
@@ -1,3 +1,18 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any, Callable
+
+
+def cache(func: Callable) -> Callable:
+    results = {}
+
+    def wrapper(*args) -> Any:
+        key = args
+        if key in results:
+            print("Getting from cache")
+            return results[key]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            results[key] = result
```

Comment:
Write this inline

Reply:
I rewrote it according to the instructions